packages/icon-button/src/IconButton.tsx

```diff
@@ -205,27 +143,102 @@ const activeStyle: { readonly [K in Variant]: string } = {
       background-color: ${uiColors.gray.dark2};
     }
   `,
-};
+} as const;
 
-const getIconStyle = (size: Size) => css`
+const iconStyle = css`
   position: absolute;
   top: 0;
   bottom: 0;
   left: 0;
   right: 0;
-  margin: auto;
-  height: ${sizeMap[size]}px;
-  width: ${sizeMap[size]}px;
+  display: flex;
+  align-items: center;
+  justify-content: center;
 `;
 
+interface IconProps extends React.SVGProps<SVGSVGElement> {
+  glyph: string;
+  size?: Size | number;
+  title?: string | null | boolean;
+}
+
+interface SharedIconButtonProps {
+  /**
+   * Determines color of `IconButton`. Can be `light` or `dark`.
+   */
+  variant?: Variant;
+
+  /**
+   * Classname applied to `IconButton`.
+   */
+  className?: string;
+
+  /**
+   * Content to appear inside of `IconButton`.
+   */
+  children?: React.ReactNode;
+
+  /**
+   * Determines whether or not `IconButton` is disabled.
+   */
+  disabled?: boolean;
```

Comment:
I gotta say the more I see JSDoc annotations used with interfaces and how it renders in VSCode the happier I am.

That being said it'd be great if we started using the [@default](https://jsdoc.app/tags-default.html) tag so we can have the annotations show the default value. Check out usePoller's `PollerOptions` to see an example.

Reply:
AFAICT TSDoc doesn't actually support default annotations. I remember exploring this when we first started doing them, and it appears to remain true.

https://github.com/Microsoft/tsdoc/issues/151

Reply:
Hmm interesting. We'd still show it in VSCode as it shows any JSDoc comments there.

Reply:
Right – we've done that for a few components before, so I suppose I can add em here.